UI/CUIControls.cpp

```diff
@@ -1924,3 +1925,136 @@ void RotatingGraphic::Render() {
     glPopClientAttrib();
     glPopMatrix();
 }
+
+////////////////////////////////////////////////
+// ToggleButton
+////////////////////////////////////////////////
+ToggleButton::ToggleButton (GG::SubTexture icon_, GG::Clr color, GG::Clr checked_clr, const std::string& tooltip_str,
```

Comment:
How is this different from StateButton?
Regardless, I suggest using StateButton or a derivate, and styling the buttons like the top-right menu bar buttons.